compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1396,13 +1396,28 @@ impl<'context> Elaborator<'context> {
             let trait_constraint_type = trait_constraint.typ.substitute(&bindings);
             let trait_bound = &trait_constraint.trait_bound;
 
+            // Replace implicitly added named generics with fresh type variables
+            let named_generics = trait_bound.trait_generics.named.iter();
+            let named_generics = named_generics.map(|named_type| {
```

Comment:
This is a bit lengthy - can you make a helper method for it instead?

Reply:
Done! I ended up with how it is because of the borrow checker (I can't pass `&trait_bound.trait_generics.named` to a helper function as it needs to take `&mut self` to create a new type variable, but `trait_bound` is a reference from self).